bitwarden_license/src/Commercial.Core/SecretManagerFeatures/Projects/DeleteProjectCommand.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Bit.Core.Exceptions;
+using Bit.Core.Repositories;
+using Bit.Core.SecretManagerFeatures.Projects.Interfaces;
+
+namespace Bit.Commercial.Core.SecretManagerFeatures.Projects;
+
+public class DeleteProjectCommand : IDeleteProjectCommand
+{
+    private readonly IProjectRepository _projectRepository;
+
+    public DeleteProjectCommand(IProjectRepository projectRepository)
+    {
+        _projectRepository = projectRepository;
+    }
+
+    public async Task<List<Tuple<Guid, string>>> DeleteProjects(List<Guid> ids)
+    {
+        var projects = await _projectRepository.GetManyByIds(ids);
+
+        if (projects?.Any() != true)
+        {
+            throw new NotFoundException();
```

Comment:
In the `BulkDeleteProjectsAsync()` function in `ProjectsController.cs`, what happens if the exception is thrown? How will it be handled? (Will the .Net Core runner handle this automatically or do we need to catch this exception?)

(same question for this line, line 22, and the line below, line 29)

Reply:
It bubbles up and is handled by the ExceptionHandlerFilterAttribute class. https://github.com/bitwarden/server/blob/sm/secrets-feature/src/SharedWeb/Utilities/ExceptionHandlerFilterAttribute.cs

This is a custom class that is used throughout the project to filter exceptions and construct return codes.

Read about custom filters here https://learn.microsoft.com/en-us/aspnet/core/mvc/controllers/filters?view=aspnetcore-6.0#exception-filters

Reply:
Perfect, thanks!